tools/libboard/board_n6010/board_n6010.c

```diff
@@ -398,20 +372,28 @@ fpga_result print_phy_info(fpga_token token)
 		opae_uio_close(&uio);
 		return res;
 	}
-	_ptr = (uint32_t *)mmap_ptr;
-	feature_list.csr = *(_ptr + HSSI_FEATURE_LIST);
+
+	feature_list.csr = *((uint32_t *) (mmap_ptr + HSSI_FEATURE_LIST));
+	hssi_ver.csr = *((uint32_t *)(mmap_ptr + HSSI_VERSION));
+
+	printf("//****** HSSI information ******//\n");
+	printf("%-32s : %d.%d  \n", "HSSI version", hssi_ver.major, hssi_ver.minor);
+	printf("%-32s : %d  \n", "Number of ports", feature_list.hssi_num);
+
 	for (i = 0; i < feature_list.hssi_num; i++) {
-		port_profile.csr = *(_ptr + HSSI_PORT_ATTRIBUTE + i);
-
-		if (port_profile.profile == HSSI_100G_PROFILE) {
-			printf("Port%d : %s \n", i, "100GCAUI-4");
-		} else if (port_profile.profile == HSSI_25G_PROFILE) {
-			printf("Port%d : %s \n", i, "25GbE");
-		} else if (port_profile.profile == HSSI_10_PROFILE) {
-			printf("Port%d : %s \n", i, "10GbE");
-		} else {
-			printf("Invalid port ");
+		port_profile.csr = *((uint32_t *)(mmap_ptr +
+			HSSI_PORT_ATTRIBUTE + i * 4));
+
+		if (port_profile.profile > HSS_PORT_PROFILE_SIZE) {
+			printf("Port%-28d :%s\n", i, "N/A");
+			continue;
 		}
+
+		for (int j = 0; j < HSS_PORT_PROFILE_SIZE; j++) {
+			if (hssi_port_profiles[j].port_index == port_profile.profile) {
+				printf("Port%-28d :%s\n", i, hssi_port_profiles[j].profile);
+			}
```

Comment:
Do you want to break out of the loop after it's found?

Reply:
I think you want break.  I think continue will just keep iterating through the inner loop.

Reply:
fixed, added break to exit for loop